internal/core/src/clustering/MemClustering.h

```diff
@@ -0,0 +1,124 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include "storage/MemFileManagerImpl.h"
+#include "storage/space.h"
+#include "clustering/Clustering.h"
+#include "knowhere/cluster/cluster_factory.h"
+
+namespace milvus::clustering {
+
+template <typename T>
+class MemClustering : public Clustering {
+ public:
+    explicit MemClustering(
+        const storage::FileManagerContext& file_manager_context =
+            storage::FileManagerContext());
+
+    void
+    Run(const Config& config) override;
+
+    ClusteringResultMeta
+    GetClusteringResultMeta() override {
+        return cluster_result_;
```

Comment:
Question: this `cluster_result_` is inited at `StreamingAssignandUpload` which is only called by `Run`. If we created a brand new `MemClustering` without calling `Run` (In DataNode), how will it be inited?

Reply:
we should check if runned